### PR TITLE
Tour Kit: Update spotlight effect, other fixes/cleanup

### LINKED ISSUE #58642: Tour Kit: Improve spotlight effect
### What

Improve the spotlight effect in the [tour-kit](https://github.com/Automattic/wp-calypso/tree/trunk/packages/tour-kit) package to enclose the whole reference element.

### Why

The spotlight effect is currently a fixed-size box that is placed over the reference element to highlight. We should make the dimension dynamic to enclose the entire reference element, potentially with some padding on the edges.

Screenshot 2021-11-30 at 1 34 50 PM

### How

_No response_

### PR DESCRIPTION
**Pull request by chriskmnds**
#### Changes proposed in this Pull Request

* Updates spotlight effect 
    * now being reference element aware (wraps the entire reference by default)
    * customizable (can be overwritten with custom CSS props e.g. in Welcome Tour)
* Updates Storybook examples
    * capture placement on all sides
    * resizable container for fiddling with scrolling etc.
* Other fixes 
    * issue with the arrow tip indicator 
    * issue with Popper misplacing the step on mount/ initial render

#### Media

Default spotlight:

Kapture 2021-12-06 at 15 30 00

Customized for Welcome Tour (fixed-size spotlight, some border radius):

Screenshot 2021-12-06 at 3 33 28 PM

#### Testing instructions

* Checkout branch, then `yarn dev --sync` from `apps/editing-toolkit` to sync the welcome tour to the sandbox
* `yarn run tour-kit:storybook:start` in another tab for Storybook demos
* Welcome Tour works as expected with/out `?welcome-tour-next`
* Stories work as expected

Closes #58642, #58748
Related to #58644

### CODE REVIEW

File: packages/tour-kit/src/components/tour-kit-frame.tsx [hunk 1]
```diff
@@ -141,11 +155,24 @@ const TourKitFrame: React.FunctionComponent< Props > = ( { config } ) => {
...
 
-	const classNames = classnames( 'tour-kit-frame', config.options?.className );
+	useEffect( () => {
+		/*
+		 * Fixes issue with Popper misplacing the instance on mount
+		 * See: https://stackoverflow.com/questions/65585859/react-popper-incorrect-position-on-mount
+		 */
+		update && update();
+		setTimeout( () => setTourReady( true ) );
+	}, [ update ] );
```
Comment (chriskmnds) [inline packages/tour-kit/src/components/tour-kit-frame.tsx]: Bug:

Kapture 2021-12-06 at 15 45 30

Fix without `tourReady` (brief flicker, not always visible):

Kapture 2021-12-06 at 15 49 47

(wish there was a better way) 😐

File: packages/tour-kit/src/components/tour-kit-frame.tsx [hunk 2]
```diff
@@ -22,8 +22,9 @@ const handleCallback = ( currentStepIndex: number, callback?: Callback ) => {
 };
 
 const TourKitFrame: React.FunctionComponent< Props > = ( { config } ) => {
+	const [ tourReady, setTourReady ] = useState( false );
 	const tourContainerRef = useRef( null );
-	const popperElementRef = useRef( null );
+	const [ popperElement, sePopperElement ] = useState< HTMLElement | null >( null );
```
Comment (chriskmnds) [inline packages/tour-kit/src/components/tour-kit-frame.tsx]: Recommended way for setting the popper element (`useState` triggers a re-render): https://popper.js.org/react-popper/v2/

Comment (escapemanuele) [reply packages/tour-kit/src/components/tour-kit-frame.tsx]: Nice one 💯

File: packages/tour-kit/src/components/tour-kit-frame.tsx (hunk 1 above)
Comment (escapemanuele) [reply packages/tour-kit/src/components/tour-kit-frame.tsx]: I made a test with the Welcome Tour and it seems that the "flickering" is still there: https://d.pr/v/24r3Xb (look when you arrive at the third point)

**Review by escapemanuele — COMMENTED** [review]
Looks very good to me, great job! 💪 
Left a comment for a  problem with `setTourReady `

File: packages/tour-kit/src/components/tour-kit-frame.tsx (hunk 1 above)
Comment (chriskmnds) [reply packages/tour-kit/src/components/tour-kit-frame.tsx]: Thanks! Yep, see it.

I think I've found a better fix 🤞 Take another look? :-)

Comment (chriskmnds) [reply packages/tour-kit/src/components/tour-kit-frame.tsx]: noticing this again while testing. looking

Comment (chriskmnds) [reply packages/tour-kit/src/components/tour-kit-frame.tsx]: Alright. Hopefully fixed now (albeit other dragons may lurk 😬 ). Give it a few more tests? :-)

This is what we should expect:

Kapture 2021-12-08 at 14 11 48

Comment (escapemanuele) [reply packages/tour-kit/src/components/tour-kit-frame.tsx]: Ok, now it seems to be working, well done!

**Review by escapemanuele — APPROVED** [review]
LGTM! After tests pass => :shipit:

File: packages/tour-kit/src/components/tour-kit-spotlight.tsx [hunk 3]
```diff
@@ -35,15 +47,30 @@ const TourKitSpotlight: React.FunctionComponent< Props > = ( { referenceElement
...
+				width: `${ referenceRect.width }px`,
+				height: `${ referenceRect.height }px`,
+		  }
+		: null;
+
 	const clipRepositionProps = referenceElement
 		? {
-				style: popperStyles?.popper,
+				style: {
+					...( clipDimensions && clipDimensions ),
```
Comment (tyxla) [inline packages/tour-kit/src/components/tour-kit-spotlight.tsx:70]: Do we really need the ` clipDimensions && ` check or can we just do`...clipDimensions`? `...null` should be fine.

File: packages/tour-kit/src/components/tour-kit-spotlight.tsx [hunk 4]
```diff
@@ -35,15 +47,30 @@ const TourKitSpotlight: React.FunctionComponent< Props > = ( { referenceElement
...
+					...popperStyles?.popper,
+					...( styles && styles ),
```
Comment (tyxla) [inline packages/tour-kit/src/components/tour-kit-spotlight.tsx:72]: Same for `styles` as `clipDimensions` above.

File: packages/tour-kit/src/components/tour-kit-spotlight.tsx (hunk 3 above)
Comment (chriskmnds) [reply packages/tour-kit/src/components/tour-kit-spotlight.tsx:70]: Thanks @tyxla. Added follow-up issue. 🙇 

(we are tracking in internal PT)

Comment (tyxla) [reply packages/tour-kit/src/components/tour-kit-spotlight.tsx:70]: Thanks, Christos 🙌

**Pull request merged**